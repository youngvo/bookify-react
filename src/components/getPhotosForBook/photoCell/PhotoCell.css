.selecting-photo-cell {
    display: inline-block;
    margin: 5px 5px 0 0;
}

.selected-photo-cell {
    display: inline-block;
    margin: 5px 5px 0 0;
}

.preview-photo {
    min-height: 75px;
    min-width: 75px;
    position: absolute;
    border: 1px hidden;
    border-radius: 3px;
    z-index: 5;
}

.preview-photo-zone {
    position: relative;
    width: 75px;
    height:75px;
    overflow: hidden;
    border:  5px solid white;
    border-radius: 5px;
    background: #fff;
}

.preview-photo-zone:hover {
    border-color: #a9e1fa;
}

.view-button-zone {}

.view-button {
    min-width: 100%;
    /*margin: 0 16px;*/
    background-image: linear-gradient(to bottom, #FFFFFF 30%, #DDDDDD 60%);
    border: 1px solid #bebebe;
    height: 25px;
    /*color: #555555;*/
    border-radius: 5px;
    margin-bottom: 3px;
    margin-top:5px;
}

.photo-cell-remove-icon {
    z-index: 15;
    margin-top: -89px;
    position: absolute;
    margin-left: 66px;
    font-size: 26px;
    cursor: pointer;
}

.selected-photo-cell:hover .icon-DeletePageIconNormal {
    display: block;
}

.selected-photo-cell:hover .preview-photo {
    border-color: #fff;
}

.icon-DeletePageIconNormal:hover span.path1 {
    text-shadow: 1.2px 0px 0px #00abe9, 0px 1.2px 0px #00abe9, 0px -1.2px 0px #00abe9, -1.2px 0px 0px #00abe9;
}

.icon-DeletePageIconNormal:hover span.path5 {
    text-shadow: 1.2px 0px 0px #00abe9, 0px 1.2px 0px #00abe9, 0px -1.2px 0px #00abe9, -1.2px 0px 0px #00abe9;
}

.icon-DeletePageIconNormal:hover span.path2 {
    -webkit-text-fill-color: #053462;
}

.icon-DeletePageIconNormal:hover span.path3 {
    -webkit-text-fill-color: #053462;
}

.icon-DeletePageIconNormal:hover span.path4 {
    -webkit-text-fill-color: #053462;
}

.view-button:hover {
    color: #00abe9;
}

.modal-wrapper {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: column;
    -webkit-flex-direction: column;
    justify-content: center;
    color: white;
    background: rgba(55, 55, 55, 0.3);
    z-index: 2;
}

.content-modal {
    align-self: center;
    display: inline-flex;
    position: absolute;
    background-image: linear-gradient(to bottom, #FFFFFF 30%, #DDDDDD 90%);
    border-radius: 10px;
    padding: 4px;
    box-shadow: 5px 5px 30px black;
}

.modal-item {
    color: gray;
    min-height: 200px;
    min-width: 200px;
    position: relative;
    display: block;
    align-self: center;
    transition: opacity 0.5s ease-in;
}

.content-modal .modal-button-item {
    z-index: 3;
    position: absolute;
    height: 20px;
    width: 20px;
    text-align: center;
    color: aqua;
}

.close-button {
    right: 4px;
}

.info-button {
    right: 24px;
}

.info-item {
    text-align: center;
    justify-content: center;
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    background-image: linear-gradient(to bottom, #FFFFFF 30%, #DDDDDD 90%);
}

.info-item-zone {
    display: inline-block;
    justify-content: center;
    top: 50%;
    transform: translate(0, -50%);
    position: relative;
}


/*effect for info div*/

.the-info {
    margin-top: 2em;
    font-size: 4em;
}

.info-enter {
    opacity: 0.01;
    transition: opacity 1s ease-in;
}

.info-enter.info-enter-active {
    opacity: 1;
}

.info-leave {
    opacity: 1;
    transition: opacity 1s ease-in;
}

.info-leave.info-leave-active {
    opacity: 0.01;
}

.selecting-photo-cell.photo-cell-selected {
    border-color: yellow;
}

