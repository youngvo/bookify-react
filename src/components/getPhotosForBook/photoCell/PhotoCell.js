import React, { Component } from 'react';

import './PhotoCell.css';
import LocaleUtils from './../../../utils/LocaleUtils';
import PhotoDetail from './../../bookDesign/photoDetail/PhotoDetail';

class PhotoCell extends Component{
    constructor(props) {
        super(props);
        this.state = {
            isOpenPhotoDetail: false,
            isHovering: false,
        };
    }

    onHover() {
        this.setState({
            isHovering: !this.state.isHovering
        });
    }

    onPhotoClick(photo, onClick) {
        let isSelected = !photo.isSelected;
        if (onClick) onClick(photo, isSelected);
    }

    closePhotoDetail() {
        this.setState({ isOpenPhotoDetail: false });
    };

    showPhotoDetail() {
        this.setState({ isOpenPhotoDetail: true });

    };

    canClick(onClick) {
        let result = true;
        if (!onClick) result = false;

        return result;
    }

    renderPicture(photo, isSelected, onClick, belongSelected) {
        let imageStyle = { background: '#ffffff' };

        if (this.state.isHovering && this.canClick()) {
            if (belongSelected) {
                imageStyle.borderColor = '#fff';
            } else {
                imageStyle.borderColor = '#a9e1fa';
            }
        }

        if (isSelected && !belongSelected) {
            imageStyle.borderColor = '#EC6E1A'
        }

        return (
            <div className="preview-photo-zone"
                style={imageStyle}
                onClick={this.canClick(onClick) && this.onPhotoClick.bind(this, photo, onClick)}>

                <img className="preview-photo" onDoubleClick={this.showPhotoDetail.bind(this)} src={photo.picUrl} alt="Detail" />
            </div>
        );
    }

    renderRemoveIcon(isHovering, isSelected, photo, iconClick) {
        if (isHovering && isSelected) {
            return (
                <div className="photo-cell-remove-icon" >
                    <span className="icon-DeletePageIconNormal" onClick={() => {iconClick(photo, false)}}>
                        <span className="path1"></span>
                        <span className="path2"></span>
                        <span className="path3"></span>
                        <span className="path4"></span>
                        <span className="path5"></span>
                    </span>
                </div>
            );
        }
    }

    render() {
        const { isOpenPhotoDetail, isHovering } = this.state;
        const { photo, onClick, isSelected, iconClick } = this.props;
        
        let displayStyle = {
            opacity: isHovering ? 1 : 0
        }
        let cellStyle = isSelected ? "selected-photo-cell" : "selecting-photo-cell";

        return (
            <div
                className={cellStyle}
                onMouseEnter={this.onHover.bind(this)}
                onMouseLeave={this.onHover.bind(this)}
            >
                {this.renderPicture(photo, photo.isSelected, onClick, isSelected)}
                <div
                    className="view-button-zone"
                >
                    {isOpenPhotoDetail && <PhotoDetail onClose={this.closePhotoDetail.bind(this)} photo={photo} />}
                    {this.renderRemoveIcon(isHovering, isSelected, photo, iconClick)}
                    <button className="view-button" style={displayStyle} onClick={this.showPhotoDetail.bind(this)} >
                        {LocaleUtils.instance.translate('import.photos.button.view')}
                    </button>
                </div>
            </div>
        );
    }
}

export default PhotoCell;