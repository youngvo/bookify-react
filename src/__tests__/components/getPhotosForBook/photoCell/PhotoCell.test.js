import React from 'react';
import PhotoCell from './../../../../components/getPhotosForBook/photoCell/PhotoCell';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';

describe('Test PhotoCell component', () => {
    let onClick;
    let photo;
    let photoCell;

    beforeEach(() => {
        onClick = jest.fn();
        photo = {
                    isSelected: false,
                    picUrl: '',
                };

        photoCell = shallow(<PhotoCell onClick={onClick} photo={photo}/>);
    });

    describe('PhotoCell snapshot', () => {
        it('PhotoCell render right', () => {
            const component = renderer.create(<PhotoCell onClick={onClick} photo={photo}/>);
            const tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
    // describe('PhotoCell state', () => {
    //     it('PhotoCell local State', () => {
    //         expect(photoCell.instance().state.isHovering).toEqual(false);
    //         expect(photoCell.instance().state.isOpenPhotoDetail).toEqual(false);
    //     })
    // });

    describe('PhotoCell renders', () => {
        it('render button', () => {
            const button = photoCell.find('button').first();
            expect(button).toBeDefined();
            expect(button.text()).toBe('View');
        });

        it('render image', () => {
            const img = photoCell.find('img').first();
            expect(img).toBeDefined();
        });

        it('render picture', () => {
            expect(photoCell.find('preview-photo')).toBeDefined();
        })

    });

    describe('PhotoCell functions', () => {
        it('PhotoCell onHover()', () => {
            photoCell.instance().onHover();
            expect(photoCell.instance().state.isHovering).not.toEqual(false);
            photoCell.instance().onHover();
            expect(photoCell.instance().state.isHovering).toEqual(false);
        });

        it('PhotoCell closePhotoDetail()', () => {
            photoCell.instance().closePhotoDetail();
            expect(photoCell.instance().state.isOpenPhotoDetail).toEqual(false);
        });

        it('PhotoCell showPhotoDetail()', () => {
            photoCell.instance().showPhotoDetail();
            expect(photoCell.instance().state.isOpenPhotoDetail).toEqual(true);
        });

        it('PhotoCell canClick', () => {
            let onClick = true;
            let result = photoCell.instance().canClick(onClick);
            expect(result).toEqual(true);
            onClick = false;
            result = photoCell.instance().canClick(onClick);
            expect(result).not.toEqual(true);
        });

        it('PhotoCell onPhotoClick()', () => {  
            let isSelected = true;
            photoCell.instance().onPhotoClick(photo, onClick);
            expect(onClick.mock.calls.length).not.toBe(2);
        });

    });


})










































