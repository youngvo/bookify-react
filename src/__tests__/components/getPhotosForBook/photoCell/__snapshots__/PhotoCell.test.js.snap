exports[`Test PhotoCell component PhotoCell snapshot PhotoCell render right 1`] = `
<div
  className="selecting-photo-cell"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  <div
    className="preview-photo-zone">
    <img
      alt="Detail"
      className="preview-photo"
      onClick={[Function]}
      onDoubleClick={[Function]}
      src=""
      style={
        Object {
          "background": "#ffffff",
        }
      } />
  </div>
  <div
    className="view-button-zone">
    <button
      className="view-button"
      onClick={[Function]}
      style={
        Object {
          "opacity": "0",
        }
      }>
      View
    </button>
  </div>
</div>
`;
